core/trino-main/src/main/java/io/trino/FeaturesConfig.java

```diff
@@ -98,6 +100,7 @@
     private boolean legacyCatalogRoles;
     private boolean incrementalHashArrayLoadFactorEnabled = true;
     private boolean allowSetViewAuthorization;
+    private Duration materializedViewRequiredFreshness = new Duration(0, SECONDS);
```

Comment:
that most likely should be a MV property

Reply:
It should definitely be allowed as an MV property as it is much easier to think about how much staleness is acceptable for specific MVs rather than defining it globally.
In fact, I would lean against making it configurable globally at all as it seems very tricky for an admin to come up with an acceptable staleness value for all MVs. It also creates the possibility of end users unexpectedly receiving stale results without having opted into it explicitly through MV property or session property.

Reply:
Data will likely have different retention policy, therefore it has to be defined per MV

Reply:
@sopel39 @raunaqmorarka @hashhar 
mistakenly replied in a different (related) thread here: https://github.com/trinodb/trino/pull/13484#discussion_r937510261